include/picongpu/plugins/openPMD/WriteSpecies.hpp

```diff
@@ -256,25 +258,70 @@ namespace picongpu
 
             using openPMDFrameType = Frame<OperatorCreateVectorBox, NewParticleDescription>;
 
-            void setParticleAttributes(::openPMD::Iteration& iteration)
+            void setParticleAttributes(::openPMD::ParticleSpecies& record)
             {
                 const float_64 particleShape(GetShape<ThisSpecies>::type::assignmentFunctionOrder);
-                iteration.setAttribute("particleShape", particleShape);
+                record.setAttribute("particleShape", particleShape);
 
                 traits::GetSpeciesFlagName<ThisSpecies, current<>> currentDepositionName;
                 const std::string currentDeposition(currentDepositionName());
-                iteration.setAttribute("currentDeposition", currentDeposition.c_str());
+                record.setAttribute("currentDeposition", currentDeposition.c_str());
 
                 traits::GetSpeciesFlagName<ThisSpecies, particlePusher<>> particlePushName;
                 const std::string particlePush(particlePushName());
-                iteration.setAttribute("particlePush", particlePush.c_str());
+                record.setAttribute("particlePush", particlePush.c_str());
 
                 traits::GetSpeciesFlagName<ThisSpecies, interpolation<>> particleInterpolationName;
                 const std::string particleInterpolation(particleInterpolationName());
-                iteration.setAttribute("particleInterpolation", particleInterpolation.c_str());
+                record.setAttribute("particleInterpolation", particleInterpolation.c_str());
 
                 const std::string particleSmoothing("none");
-                iteration.setAttribute("particleSmoothing", particleSmoothing.c_str());
+                record.setAttribute("particleSmoothing", particleSmoothing.c_str());
+
+                // now we have a map in a writeable format with all zeroes
+                // for each record copy it and modify the copy, e.g.
+
+                // const records stuff
+                ::openPMD::Datatype dataType = ::openPMD::Datatype::DOUBLE;
+                ::openPMD::Extent extent = {0};
```

Comment:
Oh no, this is a bug @PrometheusPi @franzpoeschel 

The extent need sto be the number of particles

Reply:
I believe at the time this code was written, openPMD API onlz accepted this for the const records.

Reply:
> I believe at the time this code was written, openPMD API version used only accepted this for the const records.

I don't think that was ever the case? Specifying constant record components with arbitrary extent was always possible. I think, the opposite is the case, specifying empty datasets like this was not possible for a long time (but is possible with all versions of openPMD that PIConGPU supports).

With code above, openPMD will automatically detect that the dataset is empty and create an empty dataset via `makeEmpty()` (which could alternatively also be called explicitly).

Reply:
I think you know better regarding openPMD, but there was certainly some issue that lead to this code. Either the validator didn't accept it, or the size was allocated incorrectly for each particle. Perhaps @PrometheusPi remembers better, i think we were discussing it while sitting at his laptop

Reply:
@ax3l Thanks for finding this bug. @franzpoeschel thanks for clarifying the openMPD-api behavior.

Reply:
Is this bug already fixed?

Reply:
No, not yet

Reply:
It is not fixed, I will open a PR in few minutes.

Reply:
fixed in #3678